warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexConfig.java

```diff
@@ -0,0 +1,168 @@
+package datawave.ingest.data.config.ingest;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Container class for keeping track of configuration rules related to Whindex entries.
+ * <p>
+ * A whindex rule configuration is defined by several parameters:
+ * <ul>
+ * <li><strong>valueField</strong> - The field in which an event map's values are checked. The event map must contain a matching entry for this field in order
+ * to create a Whindex entry.</li>
+ * <li><strong>values</strong> - A list of values that need to be present in the event map's valueField to create a whindex entry.</li>
+ * <li><strong>sourceField</strong> - A field that must be present in the event map for a Whindex entry to be created. Depending on configuration, this field
+ * may be replaced with the destination field.</li>
+ * <li><strong>destField</strong> - Also known as the "WhindexField". This is the name of the new field created when a whindex entry is generated. The value of
+ * this new field will be that of the sourceField when generated.</li>
+ * <li><strong>overloaded</strong> - A boolean flag indicating whether the original source field should be deleted (true) or retained (false) once the whindex
+ * entry has been processed. Inclusion of this is optional, the default value is assumed false.</li>
+ * </ul>
+ * </p>
+ */
+public class WhindexConfig {
+
+    // The field in which the event map's values are checked.
```

Comment:
The comments in this class should not reference an 'event map'. There is no actual event map in this class, only in WhindexFieldIngestHelper. This can lead to confusion about why an event map is mentioned if someone is only looking at the documentation of this class. I would change the comments here (and elsewhere in the class) to focus more on the purpose of each member. For instance,
```java
    /**
     * The field name parsed from an instance of the property {@link WhindexFieldIngestHelper#VALUE_FIELD}.
     */
    private String valueField;

    /**
     * The field values of {@link #valueField} parsed from an instance of the property {@link WhindexFieldIngestHelper#VALUES}.  
     */
    private List<String> values;
```